packages/graphql/src/translate/where/create-node-where-and-params.ts

```diff
@@ -144,7 +148,76 @@ function createNodeWhereAndParams({
                 resultStr += recurse[0];
                 resultStr += ")"; // close NONE/ANY
                 res.clauses.push(resultStr);
-                res.params = { ...res.params, fieldName: recurse[1] };
+                res.params = { ...res.params, [fieldName]: recurse[1] };
+                return res;
+            }
+
+            if (connectionField) {
+                let nodeEntries: Record<string, any> = value;
+
+                if (!connectionField?.relationship.union) {
+                    nodeEntries = { [connectionField.relationship.typeMeta.name]: value };
+                }
+
+                Object.entries(nodeEntries).forEach((entry) => {
+                    const refNode = context.neoSchema.nodes.find((x) => x.name === entry[0]) as Node;
+                    const relationship = context.neoSchema.relationships.find(
+                        (x) => x.name === connectionField.relationshipTypeName
+                    ) as Relationship;
+
+                    const safeNodeVariable = `${nodeVariable.replace(/\./g, "_")}`;
+
+                    const relatedNodeVariable = `${safeNodeVariable}_${refNode.name}`;
+
+                    const inStr = connectionField.relationship.direction === "IN" ? "<-" : "-";
+                    const relationshipVariable = `${relatedNodeVariable}_${connectionField.relationshipTypeName}`;
+                    const outStr = connectionField.relationship.direction === "OUT" ? "->" : "-";
+                    const labels = refNode.getLabelString(context);
+
+                    const collectedMap = `${relatedNodeVariable}_map`;
+
+                    const rootParam = parameterPrefix.split(".", 1)[0];
+
+                    const existsStr = `EXISTS((${safeNodeVariable})${inStr}[:${connectionField.relationship.type}]${outStr}(${labels}))`;
+
+                    if (value === null) {
+                        res.clauses.push(not ? `NOT ${existsStr}` : existsStr);
+                        return;
+                    }
+
+                    const resultArr = [
+                        `RETURN ${existsStr}`,
+                        `AND ${
+                            not ? "NONE" : "ANY"
+                        }(${collectedMap} IN [(${safeNodeVariable})${inStr}[${relationshipVariable}:${
+                            connectionField.relationship.type
+                        }]${outStr}(${relatedNodeVariable}${labels}) | { node: ${relatedNodeVariable}, relationship: ${relationshipVariable} } ] INNER_WHERE `,
+                    ];
+
+                    const connectionWhere = createConnectionWhereAndParams({
+                        whereInput: entry[1] as any,
```

Comment:
**question**:  Is there a reason we are not using `GraphQLWhereArg` here? :eyes:
It feels like the recasting of `nodeEntries` to `Record<string, any>` may not be necessary?